src/lib/input/input.ts

```diff
@@ -229,6 +229,10 @@ export class MdInput implements ControlValueAccessor, AfterContentInit, OnChange
 
   @ViewChild('input') _inputElement: ElementRef;
 
+  public elementType: 'input' | 'textarea' = undefined;
```

Comment:
You can omit the `public` (as the default). Can you also prefix this with an underscore (which we're using with the absence of the `private` keyword as "internal").